packages/web-components/src/components/va-maintenance-banner/va-maintenance-banner.tsx

```diff
@@ -0,0 +1,150 @@
+import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
+import classNames from 'classnames';
+import {
+  formatDate,
+  isDateAfter,
+  isDateBefore,
+  isDateSameDay,
+} from '../../utils/date-utils';
+
+/**
+ * @componentName Maintenance Banner
+ * @maturityCategory use
+ * @maturityLevel deployed
+ * @guidanceHref form/maintenance-banner
+ */
+
+@Component({
+  tag: 'va-maintenance-banner',
+  styleUrl: 'va-maintenance-banner.scss',
+  shadow: true,
+})
+export class VaMaintenanceBanner {
+
+  maintenanceBannerEl: HTMLDivElement;
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics?: boolean = false;
```

Comment:
I think if we are not wiring up analytics yet, we probably shouldn't advertise that there is analytics being fired. Should we just create a new ticket for completing that task after this component is done? Here is an example issue for adding analytics to va-notification: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1803

Reply:
Good question. I am unsure what needs analytics on this component. Should there be an event for whenever a maintenance banner renders and maybe when a user dismisses it? There is not anything else that is actionable on this component.

Reply:
I'm guessing we'll want something similar to va-promo-banner or any of the components that capture a close event: https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-promo-banner/va-promo-banner.tsx#L77-L104

Reply:
I added the events, please check my work though :)

Reply:
I think the other component was using `innerContent` because everything was being passed through a slot. We have some props here though so usually we just do something like this to fine tune it:

```
        details: {
          header: this.maintenanceTitle,
          starts: this.startsAt,
          expires: this.expiresAt
        },
```

I don't know, we (DST) can capture whatever we think is valuable. For a maintenance banner, those bits seems useful-ish?

Reply:
We will also need to add the component to vets-website here: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/site-wide/component-library-analytics-setup.js

Reply:
Updated for the component, will work on vets-website PR